pkg/core/function.go

```diff
@@ -147,23 +145,21 @@ func (c *client) CreateFunction(options CreateFunctionOptions, log io.Writer) (*
 	return s, nil
 }
 
-func (c *client) buildLocally(appDir string, buildImage string, runImage string, repoName string, publish bool) error {
+func (c *client) buildLocally(appDir string, buildImage string, runImage string, repoName string) error {
 	if buildImage == "" {
 		return fmt.Errorf("unable to build function locally: buildpack image not specified")
 	}
 	if runImage == "" {
 		return fmt.Errorf("unable to build function locally: run image not specified")
 	}
 
-	noPull := strings.HasPrefix(buildImage, "dev.local") && strings.HasPrefix(runImage, "dev.local")
-
 	b, err := c.buildFactory.BuildConfigFromFlags(&pack.BuildFlags{
```

Comment:
If we don't require special values for those two flags, then we can fall back to the `Run()` method we used before and simplify the code (may mess the mocking contortions you went through recently @glyn , but should simplify it I think)

Reply:
@ericbottard Yes, thanks. I pushed an extra commit with the simplifications.

Reply:
Sorry, I meant `pack.Build()` not `Run()`, as in the reversal of what happened [here](https://github.com/projectriff/riff/commit/2d33e1d414ae9c8ca7490ccd997a0d8be7d5f77d#diff-962221b752f1f6c34ce677a7163a2875L109)  (not sure this can be directly mocked though)

Reply:
Yes - please see the latest commit for the replacement interface.